### ISSUE TITLE
Using DLL for static libs causes problems

### ISSUE THREAD

**Issue by ffulin** [ClxS/FASTBuild-UE4#1]
Thanks for making this script available for UE4 users!

It seems that some assumptions have been made about how FASTBuild works, as per the comments on the wiki:

> It uses the DLL node for all linking steps. This is because FBuild's node types aren't strictly for certain steps, but just describe behaviour. The DLL node is the only one which fits our requirements. (I think?)

This seems to be causing problems for users.  Use of DLL() makes FASTBuild behave in ways that are not appropriate for static libraries.  (As noted here: https://github.com/fastbuild/fastbuild/issues/122) What is preventing the script using Library() for static libraries as FASTBuild expects? Can I help you get it working?

**Comment by ClxS** [comment]
Hi there!
It's been quite a while since I wrote this one, and it's changed quite a bit on our internal version but this is what I remember about it.

I ended up having to use DLL nodes as the Library nodes (Unless I misunderstood the docs) need you to provide the Compiler options rather than just a bunch of LinkObjects like DLL nodes support. The issue is that Unreal just provides a list of command lines to build the objects, and a command line (with a response file) to build the library. It'd be possible to just parse these first and then just generate a proper Library node, but I was concerned that we might eventually run into a situation where individual files in the project might require different arguments. (like enabling symbols/disabling optimization)

Looking at it now much later I can see you'd probably be able to do this by generating a Library node with all files using the most common arguments as the .CompilerInputFiles, and compiling the rest separately to be used in .LibrarianAdditionalInputs.

I'm hoping to revisit this script soon and I'll definitely fix that.
